AndroidAnnotations/androidannotations-core/androidannotations/src/main/java/org/androidannotations/internal/core/handler/ContextCompatAwareResHandler.java

```diff
@@ -66,14 +64,11 @@ private boolean hasTargetMethodInContext() {
 		return hasTargetMethod(context, androidRes.getResourceMethodName());
 	}
 
-	private void createCallWithIfGuard(EComponentHolder holder, JFieldRef ref, JBlock methodBody, JFieldRef idRef) {
+	private IJExpression createCallWithIfGuard(EComponentHolder holder, JFieldRef idRef) {
 		JVar resourcesRef = holder.getResourcesRef();
-		JConditional guardIf = methodBody._if(getClasses().BUILD_VERSION.staticRef("SDK_INT").gte(getClasses().BUILD_VERSION_CODES.staticRef(minSdkPlatformName)));
```

Comment:
we could, but we then would need a way to notify [this line](https://github.com/excilys/androidannotations/pull/1457/files#diff-6eeb2ce2606461aac88c842f34e4a083R79) that we already added the assignment based on the `if` here. it could be easily done by returning `null`. do we want that? (see latest commit)